mcp_server/gitlab_tools.py

```diff
@@ -135,6 +147,29 @@ async def get_internal_rhel_branches(
         raise ToolError(f"Failed to get branches for package {package}: {ex}")
 
 
+async def clone_repository(
+    repository: Annotated[str, Field(description="Repository URL to clone")],
+    clone_path: Annotated[str, Field(description="Absolute path where to clone the repository")],
+    branch: Annotated[str, Field(description="Branch to clone")] = None,
+) -> str:
+    """
+    Clones the specified repository to the given local path.
+    Uses GitLab token for authentication if available.
```

Comment:
Even though this tool is called from code, not by a model, it is still a MCP tool that could be used by a model, and this sentence could potentially confuse it. The consumers of this tool shouldn't need to know anything about any authentication or tokens.